frontend/app/routes/$lang+/_protected+/alerts+/subscribe+/confirm.tsx

```diff
@@ -0,0 +1,109 @@
+  const userInfoToken: UserinfoToken = session.get('userInfoToken');
+  const personalInformation: PersonalInfo = await personalInformationRouteHelper.getPersonalInformation(userInfoToken, params, request, session);
+  const preferredLanguage = personalInformation.preferredLanguageId ? await getLookupService().getPreferredLanguage(personalInformation.preferredLanguageId) : undefined;
+  const clickedButton = '';
+  instrumentationService.countHttpStatus('alerts.confirm', 302);
+  return json({ csrfToken, meta, personalInformation, preferredLanguage, clickedButton });
+}
+
+export async function action({ context: { session }, params, request }: ActionFunctionArgs) {
+  const personalInformationRouteHelper = getPersonalInformationRouteHelpers();
+  const userInfoToken: UserinfoToken = session.get('userInfoToken');
+  const personalInformation: PersonalInfo = await personalInformationRouteHelper.getPersonalInformation(userInfoToken, params, request, session);
+
+  const formData = await request.formData();
+
+  if (formData.get('new-code')) {
+    //TODO EMAIL THE CODE TO THE CLIENTS ADDRESS
+    personalInformation.emailAddress;
+  }
+
+  if (formData.get('submit')) {
+    //TODO VALIDATE THE CONFIRMATION CODE
+  }
+
+  return '';
+}
+
+export default function ConfirmSubscription() {
+  const { i18n, t } = useTranslation(handle.i18nNamespaces);
+  const { preferredLanguage } = useLoaderData<typeof loader>();
+  const params = useParams();
+  const fetcher = useFetcher<typeof action>();
+  const userOrigin = useUserOrigin();
+
+  return (
+    <>
+      <fetcher.Form className="max-w-prose" method="post" noValidate>
+        <div className="mb-8 space-y-6">
+          <p id="confirmation-information" className="mb-4">
+            {t('alerts:confirm.confirmation-information-text')}
+          </p>
+          <p id="confirmation-completed" className="mb-4">
+            {t('alerts:confirm.confirmation-completed-text')}
+          </p>
+          <div className="grid gap-12 md:grid-cols-2">
+            <p id="confirmation-language" className="mb-4">
+              {t('alerts:confirm.confirmation-selected-language')}
+            </p>
+            <p>{preferredLanguage ? getNameByLanguage(i18n.language, preferredLanguage) : t('alerts:confirm.no-preferred-language-on-file')}</p>
+          </div>
+          <InputField id="confirmationCode" className="w-full" label={t('alerts:confirm.confirmation-code-label')} maxLength={100} name="confirmationCode" required />
+        </div>
+        <div className="flex flex-wrap items-center gap-3">
+          <ButtonLink id="back-button" to={userOrigin?.to} params={params}>
+            {t('alerts:confirm.back')}
+          </ButtonLink>
+          <Button id="new-code-button" name="new-code" value="new-code" variant="alternative">
+            {t('alerts:confirm.request-new-code')}
+          </Button>
+          <Button id="submit-button" name="submit" value="submit" variant="primary">
```

Comment:
**Nitpick:** We don't need to have two different `name` here to limit some of the complexity in the `action()`. 

For example, we can have `name="action"` for both buttons. 
Then in the `action()`, we can do:

```typescript
const action = formData.get('action')
if (action === 'new-code') { ... }
if (action === 'submit') { ... }
```

We should also avoid hard-coding the values by defining an `enum` of the possible values. Example: https://github.com/DTS-STN/canadian-dental-care-plan/blob/8e4f5012027931b0d00b03b255a6dc5f8d23bceb/frontend/app/routes/%24lang%2B/_public%2B/apply%2B/%24id%2B/disability-tax-credit.tsx#L26

Reply:
Done